app/src/main/java/com/otaman/notes/viewmodel/EditNoteViewModel.kt

```diff
@@ -2,16 +2,20 @@ package com.otaman.notes.viewmodel
 
 import android.app.Application
 import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.otaman.notes.model.Note
 import com.otaman.notes.model.NoteRepository
-import com.otaman.notes.model.room.NoteDatabase
+import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import javax.inject.Inject
 
-class EditNoteViewModel(application: Application): AndroidViewModel(application) {
-    private val dao = NoteDatabase.getInstance(application).getNoteDao()
-    private val repository = NoteRepository.getInstance(dao)
+@HiltViewModel
+class EditNoteViewModel @Inject constructor(
+    application: Application,
+    private val repository: NoteRepository
+): AndroidViewModel(application) {

```

Comment:
We don't need application anymore, so `AndroidViewModel` can be replaced with `ViewModel`